design-proposals/multiarch-cluster.md

```diff
@@ -0,0 +1,149 @@
+As it stands, Kubevirt has provided support for both X86 and ARM-based clusters but the assumption is that the architecture of the cluster is homogeneous. 
+
+This means that all components are compiled for the architecture where they will be running. The codebase uses the runtime package to determine the compiled architecture for applicable logic branches in the code. 
+
+For users running a mixture of VM workloads, the only currently available option is to introduce multiple clusters. The complexities associated with federating multiple clusters in unison can serve as a deterrent to the usefulness of KubeVirt in these types of use cases.
+
+### Purpose 
+This document aims to explain the justification for any changes required to KubeVirt in order to provide the ability for users to run in a multi-architecture environment.  In addition, this document aims to provide an open forum to discuss whether the proposed changes are the most effective way of providing the desired functionality.
+
+## Motivation
+
+The motivation for adding the proposed functionality stemmed from a personal use case that required running X86 and ARM-based workloads. After investigation, the conclusion reached was that no such functionality was currently available in KubeVirt. This is largely due to the lack of users working with ARM-based workloads. 
+
+Regardless of this fact, it was identified that the framework already has accomplished most of the heavy lifting by providing support for each of the architecture exclusively. Given this, the amount of effort required to expand the existing logic was identified to be minimal and a POC was created. 
+
+## Goals
+
+- Support the deployment of Kubevirt into multi-architecture clusters
+- Ability to specify architecture when deploying VM/VMIs 
+- Feature changes should not break any existing functionality in KubeVirt
+- Multi-architecture support should be placed behind a feature-gate and be an optional config item based on user-need
+
+## User Stories
+
+As a user, I want to be able to create and start one or more VM/VMIs which allow me to explicitly specify the architecture of the workload
+
+## Repos
+ - https://github.com/kubevirt/kubevirt
+
+# Design
+
+## Proposed Code Changes
+
+*In addition to the following code changes, all KubeVirt components were compiled as multi-architecture images*
+
+**Kubevirt API Spec**
+
+- Addition of an Architecture field to the VirtualMachineInstanceSpec will allow the user to indicate the architecture of the workload being deployed. If not specified, the default from the cluster config will be applied.
+- Addition of DefaultVMIArchitecture and ArchConfiguration fields to the KubevirtConfiguration. The DefaultVMIArchitecture will be applied to a VMI in the event that the user has not specified it explicitly. The ArchConfiguration allows for setting defaults for the existing arch-specific config items. These items are OVMFPath, EmulatedMachines, and MachineType
```

Comment:
You can use backticks , to highlight api fields, I think that makes it easier to understand the text. E.g. `MachineType`.